src/core/src/lib.rs

```diff
@@ -238,3 +239,107 @@ pub trait Device: Sized {
     /// Cleanup unused resources. This should be called between frames. 
     fn cleanup(&mut self);
 }
+
+
+/// An `Instance` holds per-application state for a specific backend
+pub trait Instance {
+    /// Associated `Adapter` type.
+    type Adapter: Adapter;
```

Comment:
is there a single adapter associated with the instance?
I thought the instance is the main entry point, and it can have multiple different adapters exposed.
These adapters may have different queue types/families, so must have different types.